### DISCUSSION TITLE
Ideas on how to solve a problem with dates

### DISCUSSION THREAD

**Discussion by jagjordi** [immich-app/immich#1325] [Q&A]
Hello,

I have recently moved my photo collection to immich, so far I am extremely happy with it!
However I have a problem. Some time ago I restored a backup of my phone with tens of GBs of pictures, those pictures were then eventually moved to my server. Recently when I installed immich I used the CLI to upload all the pictures from my server to the immich database. What happens now is that thousands of pictures are tagged with the wrong date, they took the date of when I restored that backup. The amount of pictures is so big that when I try to navigate to that specific day using the web interface it freezes and it never comes back. 

What can I do to solve this situation?

I have found that using exiftool I can change the date of the pictures based on the jpeg name, since all of them have the date embedded in the name (IMG_DDMMYYY or similar). I have tested that with a small set of pictures and it works fine. 

If I do that to the image folder in immich, will the dates be updated in the database? Otherwise what should I do? Can I delete all the images in a specific range with the API/CLI? If I delete my user, and create a new one, and then reupload everything will the old pictures be deleted? Or should I delete it manually?

**Comment by alextran1502** [comment]
If you change the folder in Immich, the date won't be updated in the database.

Do they all have the same date, and how many assets are we aiming to correct here?

**Reply by jagjordi** [reply]
It is hard to tell because I have not been able to find which images are, as I said if I scroll into those dates it freezes. I suspect they have the same date and slightly different hours, the date of when the backup was restored and lets say the restore process took 5 hours they will have the timestamp of the moment they were extracted form the backup. I also don't know the exact number but it is probably over 10k. I will try to investigate a bit more from the originals and report back

**Reply by jrasm91** [reply]
Do the pictures have exif with the correct date?

**Reply by jagjordi** [reply]
The pictures I uploaded to immich did not have `Date/Time Original` or `Create Date`, so immich took `File Modification Date/Time` as the date for the pictures. Since the pictures came from a recovery, they all have the same `File Modification Date/Time`.

I have now fixed all those originals, using exiftool, and now they all have the correct `Date/Time Original` .

Now the easiest way to fix it is probably to delete all the pictures that came from that backup. I have all the names of the pictures so I could do it with some script. Is it possible to use the CLI to delete pictures on the server? Alternatively, is it possible to delete the pictures in the immich folder and have those changes be "seen" by the database?

**Reply by jrasm91** [reply]
I don't think there is any way currently to do a bulk delete if specific files in immich.

**Reply by lyhnet** [reply]
If he deletes the files in the library immich would not know right? The database only has path to files (created during initial upload). It does not scan/monitor the library folder. So simply overwriting the files should be doable - especially if all files have unique names its a matter of search and replace?

**Comment by nickrunning** [comment]
I have the same issue with samsung galaxy S23 ultra,  thousands of the pictures is recovery at 2023-07-15, so the timeline in immich is steak at 2023-07-15.

**Reply by aisbergde** [reply]
If the date is contained in the file name, then you could use tools like digiKam to use these dates in filenames to change dates and to write it directly into the assest and / or into sidecar files. This way I was able to correct many assets, which have been first shown by immich at import date. immich can rescan changed metadata.

**Reply by nickrunning** [reply]
You mean use digiKam throuth WebDAV, and modify the incorrect timestamp?

**Reply by aisbergde** [reply]
My assets are on my Synology
my desktop is Windows
I use SMB, assign a network drive, and then use this network drive
![image](https://github.com/immich-app/immich/assets/9020187/ba433b32-8478-4875-be8e-f77c0cad1188)

Before I used the libraries directly on my desktop drive, this is much faster. And I used "Synology drive" to sync between my desktop and the Synology. But this sync takes also a very long time.

So, I am not yet clear, which way is better. Working with digiKam on the network share takes longer for each action. Currently, I changed some dates for some thousend assets, via networkstore - > 30 minutes, in the past on local files - seconds.

**Reply by nickrunning** [reply]
I use android immich app to sync the photos. I found immich support APIs, so I get all asset info use the following command:
`curl -L 'https://xxxx.com/api/asset' \
-H 'Accept: application/json' \
-H 'x-api-key: xxxxxxxxxxxxxxxxxxxxxxxxxxx' | jq '.[] | select(.fileCreatedAt | startswith("2023-07-15")) | [.id,.originalPath]'`
Then, I will trying use other commands to apply the correct timestamp to these assets. Hope this works...

**Reply by nickrunning** [reply]
I found it was a android system issue. I use the google Gallery app, the timeline is also incorrect. So I need to force update the metadata

**Comment by mcdruid** [comment]
> ... I get all asset info use the following command: `curl -L 'https://xxxx.com/api/asset' \ -H 'Accept: application/json' \ -H 'x-api-key: xxxxxxxxxxxxxxxxxxxxxxxxxxx' | jq '.[] | select(.fileCreatedAt | startswith("2023-07-15")) | [.id,.originalPath]'` Then, I will trying use other commands to apply the correct timestamp to these assets. Hope this works...

Thanks, this helped me fix up a set of photos with the wrong date.

Getting the list of ids using a query like above, the API call to update the timestamp for each asset was something like:

```
curl -is 'http://immich.host:2283/api/asset' -X PUT -H 'Content-Type: application/json' -H 'x-api-key: XXXXX' --data-raw '{"ids":["$id"],"dateTimeOriginal":"2010-05-05T12:00:00.000+00:00"}'
```

You could probably supply a list of ids, but I iterated over the list and sent one PUT request per asset.

You can update other attributes like `latitude` and `longitude` at the same time.

**Comment by FRAOTIAC** [comment]
Hey everyone, I ran into the same issue with messed-up dates on my photo uploads to immich. I ended up writing a [Python script](https://gist.github.com/FRAOTIAC/9b68df68bb46fcad3217ff073d406fd3) that automates the process of fixing the dates through the API.

**Comment by MichaelLebrand** [comment]
Stumbled upon this thread as well after importing a bunch of whatsapp images/videos without dates, but with the date in the title. So needed a script using that to effectively set the correct date it was taken.

Updated @FRAOTIAC's script as i think the API has changed in the mean time: https://gist.github.com/MichaelLebrand/5e9fa08e575a739a24c9011d00c54960

**Reply by Leviathan09** [reply]
> Stumbled upon this thread as well after importing a bunch of whatsapp images/videos without dates, but with the date in the title. So needed a script using that to effectively set the correct date it was taken.
> 
> Updated @FRAOTIAC's script as i think the API has changed in the mean time: https://gist.github.com/MichaelLebrand/5e9fa08e575a739a24c9011d00c54960

Sadly your script doesn't wok for me.
The response_data.json is empty and i only get the output "2025-07-31 08:09:30,525 - __main__ - INFO - Fetching data from API..."
After that nothing happens

Do i have to change something else except Host-IP and API-Key?

**Reply by Leviathan09** [reply]
> > Stumbled upon this thread as well after importing a bunch of whatsapp images/videos without dates, but with the date in the title. So needed a script using that to effectively set the correct date it was taken.
> > Updated @FRAOTIAC's script as i think the API has changed in the mean time: https://gist.github.com/MichaelLebrand/5e9fa08e575a739a24c9011d00c54960
> 
> Sadly your script doesn't wok for me. The response_data.json is empty and i only get the output "2025-07-31 08:09:30,525 - **main** - INFO - Fetching data from API..." After that nothing happens
> 
> Do i have to change something else except Host-IP and API-Key?

Okay, found the missing part.
I had to adjust the dates in line 37,38,48,49

**Reply by MichaelLebrand** [reply]
ah yeah, should have exposed that as a variable - sorry :D good to hear it works.
edit: added it
